flang/lib/Optimizer/Dialect/CUF/CUFOps.cpp

```diff
@@ -227,7 +227,9 @@ mlir::LogicalResult cuf::KernelOp::verify() {
       getLowerbound().size() != getStep().size())
     return emitOpError(
         "expect same number of values in lowerbound, upperbound and step");
-
+  if (getReduceOperands().size() != getReduceAttrs()->size())
+    return emitOpError("expect same number of values in reduce operands and "
+                       "reduce attributes");
```

Comment:
Can we add a verification that the attribute in the arrays are `ReduceAttr`

Reply:
Like this?
```
  if (getReduceAttrs()) {
    for (const auto &attr : getReduceAttrs().value()) {
      if (!mlir::isa<fir::ReduceAttr>(attr))
        return emitOpError("expect reduce attributes to be ReduceAttr");
    }
  }
```